### PR TITLE
Mark Server Packages Internal

### PR DESCRIPTION
**Pull request by anthony-murphy**
For our next major version Fluid Framework will be using explicit release tagging, and d.ts rollups to support public, beta, alpha, and internal. In preparation for that we are tagging everything internal to start, and then will carve out our other releases. Marking everything internal serves 2 purposes. 1) tags are inconsistently applied today, and it is easier to carve out a consistent set for each release type, rather than finding and fixing each miss application 2) which supports 1, inconsistent tags make tooling very hard and error prone to rollout, by making everything internal we have a consistent surface to enable our tooling upon, and this tooling will be used to inform our official carve outs.

related: 
https://github.com/microsoft/FluidFramework/pull/18409
https://github.com/microsoft/FluidFramework/pull/18408
https://github.com/microsoft/FluidFramework/pull/18411
https://github.com/microsoft/FluidFramework/pull/18412
https://github.com/microsoft/FluidFramework/pull/18410
https://github.com/microsoft/FluidFramework/pull/18413

### CODE REVIEW

**Comment by alexvy86** [comment]
I don't think we enforce tags in the server packages yet; should we do it as part of / close to this change so that new things don't start getting exported by default?

**Comment by anthony-murphy** [comment]
> I don't think we enforce tags in the server packages yet; should we do it as part of / close to this change so that new things don't start getting exported by default?

 We depend on a number of server packages via the client, so we need tagging to generate the client rollups correctly. I think this is fine, but maybe there is more to do, curious what @Josmithr thinks.

**Comment by Josmithr** [comment]
> > I don't think we enforce tags in the server packages yet; should we do it as part of / close to this change so that new things don't start getting exported by default?
> 
> We depend on a number of server packages via the client, so we need tagging to generate the client rollups correctly. I think this is fine, but maybe there is more to do, curious what @Josmithr thinks.
> 
> Overall, i don't think we need to generate rollups for the server, as the usage is quite different..

I don't think there is any fundamental issue with adding the tags and adding api-extractor to these packages, even if we aren't generating trimmed roll-ups for them. But we'll have to budget that work.

File: server/routerlicious/packages/services-client/src/test/types/validateServerServicesClientPrevious.generated.ts [hunk 1]
```diff
@@ -41,7 +41,7 @@ use_current_ClassDeclaration_BasicRestWrapper(
 declare function get_current_ClassDeclaration_BasicRestWrapper():
     TypeOnly<current.BasicRestWrapper>;
 declare function use_old_ClassDeclaration_BasicRestWrapper(
-    use: TypeOnly<old.BasicRestWrapper>);
+    use: TypeOnly<old.BasicRestWrapper>): void;
```
Comment (Josmithr) [inline server/routerlicious/packages/services-client/src/test/types/validateServerServicesClientPrevious.generated.ts:44]: I'm curious what caused these files to change. Explicit return types seem like an improvement, just not sure what triggered the change.

**Review by Josmithr — APPROVED** [review]
It isn't clear to me where the generated type-test module changes came from, but the changes look like an improvement. Still, might be worth reverting since they're presumably unrelated to the tags.

**Pull request merged**